calligator/boards.py

```diff
@@ -422,26 +430,58 @@ def fill_points(self, corners, ids=None):
             return out
 
     def detect_image(self, image, subpix=True):
+
+        ids = None
+
         if len(image.shape) == 3:
             gray = cv2.cvtColor(image, cv2.COLOR_BGR2GRAY)
         else:
             gray = image
 
         size = self.get_size()
-        ret, corners = cv2.findChessboardCorners(gray, size,
-                                                 self.DETECT_PARAMS)
-
-        if ret and subpix:
-            corners = cv2.cornerSubPix(gray, corners, (3, 3), (-1, -1),
-                                       self.SUBPIX_CRITERIA)
+        pattern_found, corners = cv2.findChessboardCorners(gray, size, self.DETECT_PARAMS)
 
-        if corners is None:
-            ids = None
-        else:
+        # Does <findChessBoardCorners()> believe that it found the pattern?
+        if pattern_found:
             ids = self.ids
+            # Do we want sub-pixel accuracy?
+            if subpix:
+                # TODO: The docs at (https://docs.opencv.org/3.4.0/dd/d1a/group__imgproc__feature.html
```

Comment:
Empirically, I think the function returns None if it fails, although that happens very rarely.
This is why the corners is None check happens after this in the original code, to account for this failure.